apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java

```diff
@@ -83,9 +83,9 @@ public void setHeader(String headerName, String headerValue, Message message) {
         try {
             message.setStringProperty(headerName, headerValue);
         } catch (MessageNotWriteableException e) {
-            logger.debug("Failed to set JMS message property due to read-only message", e);
+            logger.debug("Failed to set JMS message property {} due to read-only message", headerName, e);
         } catch (JMSException e) {
-            logger.warn("Failed to set JMS message property. Distributed tracing cannot work without that.");
+            logger.warn("Failed to set JMS message property {}.", headerName);
```

Comment:
here the name of the property will make it obvious to know the case and if it's related to distributed tracing or not.
Now this is also used to add the destination name for the receiver side.

Reply:
It's not for us, it's for the user who doesn't necessarily knows this property is related to distributed tracing. In addition, this message is still valid - if we can't set a property, distributed tracing will not work. We can say "Distributed tracing _**may**_ not work..." instead.